### ISSUE TITLE
Unbalanced dataset

### ISSUE THREAD

**Issue by nikita8** [CSCCNY/final-project-faultdetection-cnny#7]

**Comment by nikita8** [comment]
- [ ]  Custom Loss calculation based on the class weights. Consider subsampling of majority class based on the minority class. Also, try considering all the outcome and performing the weight calculation based on the total occurrence. But that didn't help much. 
- [ ] Use focal loss function
- [ ] Utilize Keras class weight option. Manually provide weightage information for each class and fed during the model fit.
- [ ] Instead of Manually providing the weightage, use sklearn to compute_class_weight and pass this as class weight during the training.

**Comment by nikita8** [comment]
Apart from the loss function, also worked on the normalization of inputs on circuit-level instead of considering all the circuits. After this change, and using the Sklearn class weight, there was an improvement in model learning.
